src/architecture/unixfs.md

```diff
@@ -0,0 +1,618 @@
+- The next three least significant bits represent `setuid`, `setgid` and the `sticky bit`
+- The remaining 20 bits are reserved for future use, and are subject to change. Spec implementations **MUST** handle bits they do not expect as follows:
+  - For future-proofing the (de)serialization layer must preserve the entire uint32 value during clone/copy operations, modifying only bit values that have a well defined meaning: `clonedValue = ( modifiedBits & 07777 ) | ( originalValue & 0xFFFFF000 )`
+  - Implementations of this spec must proactively mask off bits without a defined meaning in the implemented version of the spec: `interpretedValue = originalValue & 07777`
+
+#### `mtime`
+
+A two-element structure ( `Seconds`, `FractionalNanoseconds` ) representing the
+modification time in seconds relative to the unix epoch `1970-01-01T00:00:00Z`.
+The two fields are:
+
+1. `Seconds` ( always present, signed 64bit integer ): represents the amount of seconds after **or before** the epoch.
+2. `FractionalNanoseconds` ( optional, 32bit unsigned integer ): when specified represents the fractional part of the mtime as the amount of nanoseconds. The valid range for this value are the integers `[1, 999999999]`.
+
+Implementations encoding or decoding wire-representations MUST observe the following:
+
+- An `mtime` structure with `FractionalNanoseconds` outside of the on-wire range
+  `[1, 999999999]` is **not** valid. This includes a fractional value of `0`.
+  Implementations encountering such values should consider the entire enclosing
+  metadata block malformed and abort processing the corresponding DAG.
+- The `mtime` structure is optional - its absence implies `unspecified`, rather
+  than `0`
+- For ergonomic reasons a surface API of an encoder MUST allow fractional 0 as
+  input, while at the same time MUST ensure it is stripped from the final structure
+  before encoding, satisfying the above constraints.
+
+Implementations interpreting the mtime metadata in order to apply it within a
+non-IPFS target MUST observe the following:
+
+- If the target supports a distinction between `unspecified` and `0`/`1970-01-01T00:00:00Z`,
+  the distinction must be preserved within the target. E.g. if no `mtime` structure
+  is available, a web gateway must **not** render a `Last-Modified:` header.
+- If the target requires an mtime ( e.g. a FUSE interface ) and no `mtime` is 
+  supplied OR the supplied `mtime` falls outside of the targets accepted range:
+  - When no `mtime` is specified or the resulting `UnixTime` is negative:
+    implementations must assume `0`/`1970-01-01T00:00:00Z` (note that such values
+    are not merely academic: e.g. the OpenVMS epoch is `1858-11-17T00:00:00Z`)
+  - When the resulting `UnixTime` is larger than the targets range ( e.g. 32bit
+    vs 64bit mismatch) implementations must assume the highest possible value
+    in the targets range (in most cases that would be `2038-01-19T03:14:07Z`)
```

Comment:
```suggestion
Implementations interpreting the `mtime` metadata in order to apply it within a
non-IPFS target MUST observe the following:

- If the target supports a distinction between `unspecified` and `0`/`1970-01-01T00:00:00Z`,
  the distinction must be preserved within the target. For example, if no `mtime` structure
  is available, a web gateway must **not** render a `Last-Modified:` header.
- If the target requires an `mtime` ( e.g. a FUSE interface ) and no `mtime` is 
  supplied OR the supplied `mtime` falls outside of the targets accepted range:
  - When no `mtime` is specified or the resulting `UnixTime` is negative:
    implementations must assume `0`/`1970-01-01T00:00:00Z` (note that such values
    are not merely academic: e.g. the OpenVMS epoch is `1858-11-17T00:00:00Z`)
  - When the resulting `UnixTime` is larger than the targets range ( e.g. 32bit
    vs 64bit mismatch), implementations must assume the highest possible value
    in the targets range. In most cases, this would be `2038-01-19T03:14:07Z`.
```